scripts/dts/extract_dts_includes.py

```diff
@@ -672,51 +672,62 @@ def print_key_value(k, v, tabstop):
     else:
         tabs = (len(label) >> 3) + 1
 
-    sys.stdout.write(label)
+    line = label
     for i in range(0, tabstop - tabs + 1):
-        sys.stdout.write('\t')
-    sys.stdout.write(str(v))
-    sys.stdout.write("\n")
+        line += '\t'
+    line += str(v)
+    line += '\n'
 
-    return
+    return line
 
 
-def generate_keyvalue_file(defs, args):
+def generate_keyvalue_file(defs, kv_file):
+    try:
+        fd = open(kv_file, "w")
+    except:
```

Comment:
One should filter on the exception type that is being expected.

Also, I'm not sure if the new Exception actually is an improvement on what open() would have thrown.

```
>>> open("some_file.txt", "w")
<open file 'some_file.txt', mode 'w' at 0x7f6925307540>
>>> open("a/path/to/some_file.txt", "w")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IOError: [Errno 2] No such file or directory: 'a/path/to/some_file.txt'

```

Reply:
I copy/pasted what was done with the dts file, but indeed better let the system's exception going instead of a custom one.